### PR TITLE
Feature/3361 allow ignore mutes for unread count

### LINKED ISSUE #3361: Allow ignoring of channel mutes when updating unread count
### Context & proposal

Some of our customers might want us to ignore channel mutes for the unread count. This serves the purpose of social media chats where even if a chat is muted, you might want to know what the unread count is.

### Acceptance criteria

To accept this task we need to complete the following items:

- [ ] Expose a way to customize the unread count behavior.
- [ ] Apply the flag to unread count updates.

### PR DESCRIPTION
**Pull request by filbabic**
### 🎯 Goal
Some of our customers want to support social media like use cases where muted channels receive unread count events.

This is required to support it. Fixes #3361 

### 🛠 Implementation details

Added a flag in the OP `Config` that affects the unread count logic.

### 🧪 Testing

1. Run the app on two devices on this branch with the applied patch.
2. Log in to two different users who are in the same channel.
3. Mute a channel with one user and send a message with another. The unread count should update.
4. You can also test by putting the flag to false (default behavior) that the unread count doesn't update when the channel is muted.

<details>
  <summary>Patch file to apply these changes for testing</summary>

```
Index: stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt
--- a/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt	(revision 60665b59092afdb9b5238a1c7c7fe90401a0edae)
+++ b/stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/application/ChatInitializer.kt	(date 1650444246174)
@@ -67,7 +67,7 @@
             config = Config(
                 userPresence = true,
                 persistenceEnabled = true,
-                ignoreMutesForUnreadCount = false
+                ignoreMutesForUnreadCount = true
             ),
             appContext = context
         )
Index: stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChatHelper.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChatHelper.kt b/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChatHelper.kt
--- a/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChatHelper.kt	(revision 60665b59092afdb9b5238a1c7c7fe90401a0edae)
+++ b/stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChatHelper.kt	(date 1650440259516)
@@ -58,7 +58,7 @@
             config = Config(
                 userPresence = true,
                 persistenceEnabled = true,
-                ignoreMutesForUnreadCount = false
+                ignoreMutesForUnreadCount = true
             ),
             appContext = context
         )

```

</details>

#### General
- [x] Assigned a person / code owner group (required)
- [x] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [x] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/configuration/Config.kt [hunk 1]
```diff
@@ -25,10 +25,12 @@ import io.getstream.chat.android.offline.model.message.attachments.UploadAttachm
...
  * @param persistenceEnabled Whether the data should be stored in the DB.
  * @param uploadAttachmentsNetworkType An enumeration of various network types used as a constraint inside upload attachments worker.
+ * @property ignoreMutesForUnreadCount If we should ignore muted channel state when updating the unread count.
  */
 public data class Config(
     public val backgroundSyncEnabled: Boolean = true,
     public val userPresence: Boolean = true,
     public val persistenceEnabled: Boolean = true,
     public val uploadAttachmentsNetworkType: UploadAttachmentsNetworkType = UploadAttachmentsNetworkType.NOT_ROAMING,
+    public val ignoreMutesForUnreadCount: Boolean = false
```
Comment (adasiewiczr) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/configuration/Config.kt:35]: What do you think about marking it with `@ExperimentalStreamChatAPI`?
The flag should be replaced by the config field at some point

File: stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/logic/channel/internal/ChannelLogic.kt [hunk 2]
```diff
@@ -114,13 +114,15 @@ import kotlin.math.max
...
  * @property attachmentUrlValidator [AttachmentUrlValidator] A validator to validate attachments' url.
+ * @property ignoreMutesForUnreadCount If we should ignore muted channel state when updating the unread count.
  */
 internal class ChannelLogic(
     private val mutableState: ChannelMutableState,
     private val globalMutableState: GlobalMutableState,
     private val repos: RepositoryFacade,
     private val userPresence: Boolean,
     private val attachmentUrlValidator: AttachmentUrlValidator = AttachmentUrlValidator(),
+    private val ignoreMutesForUnreadCount: Boolean = false
```
Comment (adasiewiczr) [inline stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/plugin/logic/channel/internal/ChannelLogic.kt:125]: Do we need this default value here?

**Review by adasiewiczr — CHANGES_REQUESTED** [review]
The flag is going to be included in Channel's `Config`. Let's wait until feature is implemented on the BE

**Comment by filbabic** [comment]
Blocked by BE, closing to revisit once it's ready

**Pull request closed without merging**